internal/ingress/annotations/globalratelimit/main.go

```diff
@@ -32,6 +34,46 @@ import (
 
 const defaultKey = "$remote_addr"
 
+const (
+	globalRateLimitAnnotation             = "global-rate-limit"
+	globalRateLimitWindowAnnotation       = "global-rate-limit-window"
+	globalRateLimitKeyAnnotation          = "global-rate-limit-key"
+	globalRateLimitIgnoredCidrsAnnotation = "global-rate-limit-ignored-cidrs"
+)
+
+var globalRateLimitAnnotationConfig = parser.Annotation{
+	Group: "ratelimit",
+	Annotations: parser.AnnotationFields{
+		globalRateLimitAnnotation: {
+			Validator:     parser.ValidateInt,
+			Scope:         parser.AnnotationScopeIngress,
+			Risk:          parser.AnnotationRiskLow,
+			Documentation: `This annotation configures maximum allowed number of requests per window`,
+		},
+		globalRateLimitWindowAnnotation: {
+			Validator:     parser.ValidateDuration,
+			Scope:         parser.AnnotationScopeIngress,
+			Risk:          parser.AnnotationRiskLow,
+			Documentation: `Configures a time window (i.e 1m) that the limit is applied`,
+		},
+		globalRateLimitKeyAnnotation: {
+			Validator: parser.ValidateRegex(*parser.NGINXVariable, true),
+			Scope:     parser.AnnotationScopeIngress,
+			Risk:      parser.AnnotationRiskHigh,
```

Comment:
I think it is indeed a crucial configuration, but for the sake of risk, I suggest setting it to medium.

Reply:
The bigger problem here is leaking some undesired nginx variables. I think we can set it as high and maybe get some feedbacks, wdyt?

Reply:
Makes sense, I agree.